agents/postgres/pgstatmonitor/pgstatmonitor.go

```diff
@@ -331,8 +266,80 @@ func (m *PGStatMonitorQAN) Run(ctx context.Context) {
 	}
 }
 
+type settings map[string]*pgStatMonitorSettingsTextValue
+
+func (m *PGStatMonitorQAN) getSettings() (settings, error) {
+	var settingsRows []reform.Struct
+
+	settingsValuesAreText, err := areSettingsTextValues(m.q)
+	if err != nil {
+		return nil, err
+	}
+	if settingsValuesAreText {
+		settingsRows, err = m.q.SelectAllFrom(pgStatMonitorSettingsTextValueView, "")
+	} else {
+		settingsRows, err = m.q.SelectAllFrom(pgStatMonitorSettingsView, "")
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get settings")
+	}
+
+	settings := make(settings)
+	for _, row := range settingsRows {
+		if settingsValuesAreText {
+			setting := row.(*pgStatMonitorSettingsTextValue)
+			settings[setting.Name] = setting
+		} else {
+			setting := row.(*pgStatMonitorSettings)
+			name := setting.Name
+			settings[name] = &pgStatMonitorSettingsTextValue{
+				Name:  name,
+				Value: fmt.Sprintf("%d", setting.Value),
+			}
+		}
+	}
+
+	return settings, nil
+}
+
+func (s settings) getNormalizedQueryValue() (bool, error) {
+	key := "pg_stat_monitor.pgsm_normalized_query"
+	if _, ok := s[key]; !ok {
+		return false, errors.New("failed to get pgsm_normalized_query property")
+	}
+
+	if s[key].Value == "yes" {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (s settings) getWaitTime() (time.Duration, error) {
+	key := "pg_stat_monitor.pgsm_bucket_time"
+	if _, ok := s[key]; !ok {
+		return defaultWaitTime, errors.New("failed to get pgsm_bucket_time, wait time set on 60 seconds")
+	}
+
+	valueInt, err := strconv.ParseInt(s[key].Value, 10, 64)
+	if err != nil {
+		return defaultWaitTime, errors.Wrap(err, "property pgsm_bucket_time cannot be parsed as integer, wait time set on 60 seconds")
+	}
+
+	return time.Duration(valueInt) * time.Second, nil
+}
+
 func (m *PGStatMonitorQAN) getNewBuckets(ctx context.Context, periodLengthSecs uint32) ([]*agentpb.MetricsBucket, error) {
-	current, prev, err := m.monitorCache.getStatMonitorExtended(ctx, m.q, m.pgsmNormalizedQuery)
+	settings, err := m.getSettings()
+	if err != nil {
+		m.l.Errorf(err.Error())
```

Comment:
If we get an error here, shouldn't we stop the method and return an error?
The same below

Reply:
Good catch. Moved before line 245 in pgstatmonitor.go as you required above and added continue there.